From df797b83511cbcd0c54f84b68e26a7b182671123 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Fri, 4 Aug 2017 01:24:35 +0800
Subject: [PATCH 2317/5242] MLK-16135 usb: gadget: utp: bulk out req amount
 divisible by 512

commit  0c9ed6d19b95038c5f9aa749ad3109f0a9f73203 from
https://source.codeaurora.org/external/imx/linux-imx.git

As the code comments say, the bulk out req amount should be divisible
by 512, fix this by using set_bulk_out_req_length().

Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/gadget/function/fsl_updater.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/usb/gadget/function/fsl_updater.c b/drivers/usb/gadget/function/fsl_updater.c
index f584b15..97860fa 100644
--- a/drivers/usb/gadget/function/fsl_updater.c
+++ b/drivers/usb/gadget/function/fsl_updater.c
@@ -308,8 +308,7 @@ static int utp_do_write(struct fsg_dev *fsg, void *data, size_t size)
 
 			/* amount is always divisible by 512, hence by
 			 * the bulk-out maxpacket size */
-			bh->outreq->length = bh->bulk_out_intended_length =
-					amount;
+			set_bulk_out_req_length(fsg->common, bh, amount);
 			bh->outreq->short_not_ok = 1;
 			start_transfer(fsg, fsg->bulk_out, bh->outreq);
 			fsg->common->next_buffhd_to_fill = bh->next;
-- 
1.7.9.5

